File: Julia/math/automorphic_number.jl
"""Julia program to check if a number is automorphic or not.
An Automorphic Number is such a number whose square ends in the same digits as  the number itself."""


function check_automophic(num)
    temp = num * num

    while(num > 0)
        # Extract the last digits of the given number and its square.
        lastNum = num % 10
        lastSquare = temp % 10

        # Check if they are equal
        if(lastNum != lastSquare)
            return "The given number $n is not an Automorphic Number."
        end

        num = num ÷ 10
        temp = temp ÷ 10
    end

    return "The given number $n is an Automorphic Number."
end

print("Enter the number: ")
n = readline()
n = parse(Int, n)
res = check_automophic(abs(n))
print(res)


"""
Time Complexity: O(log(n)), where 'n' is the given number
Space Complexity: O(1)

SAMPLE INPUT AND OUTPUT

SAMPLE 1

Enter the number: 24
The given number 24 is not an Automorphic Number.

SAMPLE 2

Enter the number: 25
The given number 25 is an Automorphic Number.
"""

File: Dart/maths/automorphic_number.dart
/**
 * Given a number and you need to check if a number is Automorphic or not. 
 * A number is called Automorphic if the square of the number ends
 * with the number itself.
 * Input:
 * First line of input contains a number of integer data type.
 * Output:
 * A single line telling whether a number is Automorphic or not.
 *
*/

import 'dart:io';
import 'dart:math';

int Check_Automorphic(int n) {
  int square, temp, remainder, no_digits = 0;
  temp = n;
  square = n * n;
  int flag = 10;

  while (n != 0) {
    n = (n / 10).floor();
    no_digits++;
  }

  flag = pow(10, no_digits);
  remainder = square % flag;

  if (remainder == temp)
    return 1;
  else
    return 0;
}

void main() {
  int num;
  print("Enter the number: ");
  num = int.parse(stdin.readLineSync());

  int result = Check_Automorphic(num);

  if (result == 1)
    print("$num is a Automorphic number");
  else
    print("$num is not a Automorphic number");
}

/*
 * Example:
 * Input:
 * 3
 * Output:
 * 3 is not a Automorphic number.
 * 
 * Input:
 * 25
 * Output:
 * 25 is a Automorphic number.
*/
/*
*Time complexity : O(n) 
*Space complexity : O(1)
*/
